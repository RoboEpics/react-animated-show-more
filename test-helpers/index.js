import renderer from 'react-test-renderer';

export const renderSnapshot = (component) => {
  const tree = renderer
    .create(component)
    .toJSON();
  return tree;
};
