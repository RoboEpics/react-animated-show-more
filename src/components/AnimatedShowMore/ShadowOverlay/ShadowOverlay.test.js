import React from 'react';
import { ShadowOverlay } from './index';
import { renderSnapshot } from '../../../../test-helpers';

describe('ShadowOverlay', () => {
  describe('Snapshots', () => {
    it('renders the default shadow', () => {
      const snap = renderSnapshot(<ShadowOverlay />);

      expect(snap).toMatchSnapshot();
    });

    it('renders a custom shadow', () => {
      const snap = renderSnapshot(<ShadowOverlay color="#f39" />);

      expect(snap).toMatchSnapshot();
    });
  });
});

