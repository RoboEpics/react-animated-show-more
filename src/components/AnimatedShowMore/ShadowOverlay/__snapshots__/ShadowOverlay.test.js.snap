// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShadowOverlay Snapshots renders a custom shadow 1`] = `
<div
  className="AnimatedShowMore__ShadowOverlay"
  style={
    Object {
      "backgroundImage": "linear-gradient(to bottom, transparent, #f39)",
      "bottom": 0,
      "height": 60,
      "left": 0,
      "position": "absolute",
      "width": "100%",
    }
  }
/>
`;

exports[`ShadowOverlay Snapshots renders the default shadow 1`] = `
<div
  className="AnimatedShowMore__ShadowOverlay"
  style={
    Object {
      "backgroundImage": "linear-gradient(to bottom, transparent, #fff)",
      "bottom": 0,
      "height": 60,
      "left": 0,
      "position": "absolute",
      "width": "100%",
    }
  }
/>
`;
